PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs

```diff
@@ -92,17 +101,81 @@ public DoctorAppointment RecommendAnAppointment(DoctorUser doctor, DateTime date
             return null;
         }
 
+        private List<DoctorAppointment> GetAllAppointmentsByDateAndDoctor(DateTime date, int doctorId)
+        {
+            List<DoctorAppointment> appointments = new List<DoctorAppointment>();
+            foreach (DoctorAppointment appointment in GetAppointmentsForDoctor(doctorId))
+            {
+                if(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date))
+                {
+                    appointments.Add(appointment);
+                }
+            }
+            return appointments;
+        }
+
+        private List<DoctorAppointment> GetAllAppointmentsByDateAndPatient(DateTime date, int patientId)
+        {
+            List<DoctorAppointment> appointments = new List<DoctorAppointment>();
+            foreach (DoctorAppointment appointment in GetAppointmentsForPatient(patientId))
+            {
+                if (date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date))
+                {
+                    appointments.Add(appointment);
+                }
+            }
+            return appointments;
```

Comment:
Refactor code. Replace method GetAllAppointmentsByDateAndPatient(DateTime date, int patientId) with Where syntax, for example :
return GetAppointmentsForPatient(patientId).Where(appointment =>(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date)) );